### PR TITLE
[Component with its own class name as service type] LPS-198933 Refactor DTOConverter from headless-commerce-admin-catalog-impl and headless-delivery-impl components

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-198933](https://liferay.atlassian.net/browse/LPS-198933)

Previous Pull: https://github.com/liferay-core-infra/liferay-portal/pull/5834

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:test:commerce

**Comment by zhangyan0701** [comment]
ci:test:lima-headless

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsResourceImpl.java [hunk 1]
```diff
@@ -80,8 +80,10 @@ private ProductVirtualSettings _toProductVirtualSettings(
 	@Reference
 	private CPDefinitionService _cpDefinitionService;
 
-	@Reference
-	private ProductVirtualSettingsDTOConverter
+	@Reference(
+		target = "(dto.class.name=com.liferay.headless.commerce.admin.catalog.dto.v1_0.ProductVirtualSettings)"
+	)
```
Comment (rafaprax) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductVirtualSettingsResourceImpl.java:85]: use the component.name property instead

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:startup

**Comment by RenanUmV** [comment]
ci:test:commerce

**Comment by RenanUmV** [comment]
ci:test:lima-headless

**Comment by Kvale1733** [comment]
@RenanUmV No regressions found.

**Review by rafaprax — APPROVED** [review]

**Review by dantewang — CHANGES_REQUESTED** [review]
@RenanUmV @rafaprax Code looks good, but please extract the changes that fixes the `dto.class.name` to a separate commit for better readability. Then send to Brian as there's no code change compared to this PR.

**Comment by RenanUmV** [comment]
resent here: https://github.com/liferay-core-infra/liferay-portal/pull/6254

**Pull request closed without merging**